components/fs-data-provider/src/manifest.rs

```diff
@@ -0,0 +1,38 @@
+use serde::{Deserialize, Serialize};
+
+#[non_exhaustive]
+#[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum AliasOption {
+    /// Do not de-duplicate data.
+    NoAliases,
+    /// De-duplicate data by using filesystem symlinks.
+    Symlink,
+    // TODO: Alias based on a field in the JSON file
+}
+
+#[non_exhaustive]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum SyntaxOption {
+    /// Serialize using JavaScript Object Notation (JSON).
+    Json,
+    // Future: Consider adding a custom format option here.
+    // Custom {
+    //     file_extension: String,
+    // }
+}
+
+impl SyntaxOption {
+    /// Gets the file extension associated with the given syntax.
+    pub fn get_file_extension(&self) -> &str {
+        match self {
+            SyntaxOption::Json => "json",
+        }
+    }
+}
+
+#[non_exhaustive]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub(crate) struct Manifest {
```

Comment:
Naive question: are manifests like *.lock files?  Is that the general term for those kinds of things?  Here, for ICU4X, we're serializing/exporting them as a file, right?

Reply:
I got the name from [manifest.json in the WebExtensions spec](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json).  It holds metadata about the package.